File: BookManagementSystem/Book.java
package BookManagementSystem;

/*
    Class Book must have the following attributes:

    title: a String representing the title of the book.
    author: a String representing the author of the book.
    isbn: a String that represents the book's identification number.
    price: a double representing the price of the book.
*/

public class Book {
    protected String title;
    protected String author;
    protected String isbn;
    protected double price;

    // constructor
    public Book(String title, String author, String isbn, double price){
        this.title = title;
        this.author = author;
        this.isbn = isbn;
        this.price = price;
    }   

    // method to calculate the total price
    public double calculatePriceTotal() {
        return price;
    }

    // getters and setters

    // get and set title
    public String getTitle(){
        return title;
    }
    public void setTitle(String title){
        this.title = title;
    }

    // get and set author
    public String getAuthor(){
        return author;
    }
    public void setAuthor(String author){
        this.author = author;
    }

    // get and set isbn
    public String getIsbn(){
        return isbn;
    }
    public void setIsbn(String isbn){
        this.isbn = isbn;
    }

    // get and set price
    public double getPrice(){
        return price;
    }
    public void setPrice(double price){
        this.price = price;
    }
}

File: BMSRust/src/book.rs
#[warn(non_snake_case)]

// Define a Book struct
pub struct Book {
    title: String,
    author: String,
    isbn: String,
    price: f64,
}

// Implement methods for Book
impl Book {
    pub fn new(title: String, author: String, isbn: String, price: f64) -> Book {
        Book {
            title,
            author,
            isbn,
            price,
        }
    }

    pub fn calculate_price_total(&self) -> f64 {
        self.price
    }

    // Getters and setters for title, author, isbn, and price
    pub fn get_title(&self) -> String {
        self.title.clone()
    }

    pub fn set_title(&mut self, title: String) {
        self.title = title;
    }

    pub fn get_author(&self) -> String {
        self.author.clone()
    }

    pub fn set_author(&mut self, author: String) {
        self.author = author;
    }

    pub fn get_isbn(&self) -> String {
        self.isbn.clone()
    }

    pub fn set_isbn(&mut self, isbn: String) {
        self.isbn = isbn;
    }

    pub fn get_price(&self) -> f64 {
        self.price
    }

    pub fn set_price(&mut self, price: f64) {
        self.price = price;
    }
}

// Define a PhysicalBook struct
pub struct PhysicalBook {
    book: Book,
    weight: f64,
    shipping: f64,
}

// Implement methods for PhysicalBook
impl PhysicalBook {
    pub fn new(title: String, author: String, isbn: String, price: f64, weight: f64, shipping: f64) -> PhysicalBook {
        PhysicalBook {
            book: Book::new(title, author, isbn, price),
            weight,
            shipping,
        }
    }

    pub fn calculate_price_total(&self) -> f64 {
        self.book.price + self.shipping
    }

    pub fn calculate_weight_total(&self) -> f64 {
        self.weight
    }

    pub fn calculate_shipping_total(&self) -> f64 {
        self.shipping
    }

    // Getters and setters for weight and shipping
    pub fn get_weight(&self) -> f64 {
        self.weight
    }

    pub fn set_weight(&mut self, weight: f64) {
        self.weight = weight;
    }

    pub fn get_shipping(&self) -> f64 {
        self.shipping
    }

    pub fn set_shipping(&mut self, shipping: f64) {
        self.shipping = shipping;
    }
}
